packages/investor-yearn/src/api/vaults.ts

```diff
@@ -14,27 +14,21 @@ export type SupportedYearnVault = {
   chain: ChainTypes
   provider: string
   type: string
+  expired: boolean
 }
 
 export const getSupportedVaults = async (): Promise<SupportedYearnVault[]> => {
   const vaults = await yearnSdk.vaults.get()
-
-  return vaults
-    .filter((vault: Vault) => {
-      // Currently filtering out all vaults where deposits are disabled or the deposit limit is 0.
-      // TODO: Update modal so that vaults with disabled deposit only show if a user has a balance
-      // and only display the option to withdraw from the vault.
-      return !vault.metadata.depositsDisabled && bnOrZero(vault.metadata.depositLimit).gt(0)
-    })
-    .map((vault: Vault) => {
-      return {
-        vaultAddress: toLower(vault.address),
-        name: `${vault.name} ${vault.version}`,
-        symbol: vault.symbol,
-        tokenAddress: toLower(vault.token),
-        chain: ChainTypes.Ethereum,
-        provider: DefiProvider.Yearn,
-        type: DefiType.Vault
-      }
-    })
+  return vaults.map((vault: Vault) => {
+    return {
+      vaultAddress: toLower(vault.address),
+      name: `${vault.name} ${vault.version}`,
+      symbol: vault.symbol,
+      tokenAddress: toLower(vault.token),
+      chain: ChainTypes.Ethereum,
+      provider: DefiProvider.Yearn,
+      type: DefiType.Vault,
+      expired: !!vault.metadata.depositsDisabled || bnOrZero(vault.metadata.depositLimit).lte(0)
```

Comment:
Do we need to coerce `vault.metadata.depositsDisabled`? It's already of type `boolean | undefined`.

Reply:
fixed, I was actually coercing it before as I was only using `vault.metadata.depositsDisabled` then alter on I also added the check for `vault.metadata.depositLimit`